src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -0,0 +1,2458 @@
+        barGraphs[currentBar].barType = value;
+      }
+
+      if (definitelyGreaterThan(barGraphs[currentBar].min, barGraphs[currentBar].max)) {
+        std::swap(barGraphs[currentBar].min, barGraphs[currentBar].max);
+      }
+    }
+    #  ifdef P104_DEBUG
+
+    if (logAllText && loglevelActiveFor(LOG_LEVEL_INFO)) {
+      String log;
+      log.reserve(70);
+      log = F("dotmatrix: Bar-graph: ");
+
+      if (loop) {
+        log += currentBar;
+        log += F(" in: ");
+        log += graphpart;
+        log += F(" value: ");
+        log += barGraphs[currentBar].value;
+        log += F(" max: ");
+        log += barGraphs[currentBar].max;
+        log += F(" min: ");
+        log += barGraphs[currentBar].min;
+        log += F(" dir: ");
+        log += barGraphs[currentBar].direction;
+        log += F(" typ: ");
+        log += barGraphs[currentBar].barType;
+      } else {
+        log += F(" bsize: ");
+        log += barGraphs.size();
+      }
+      addLog(LOG_LEVEL_INFO, log);
+    }
+    #  endif // ifdef P104_DEBUG
+    currentBar++; // next
+    delay(0); // Leave some breathingroom
+  }
+  #  undef NOT_A_COMMA
+
+  if (barGraphs.size() > 0) {
+    uint8_t  barWidth = 8 / barGraphs.size(); // Divide the 8 pixel width per number of bars to show
+    int16_t  pixTop, pixBottom;
+    uint16_t zeroPoint;
+    #  ifdef P104_DEBUG
+    String log;
+
+    if (logAllText &&
+        loglevelActiveFor(LOG_LEVEL_INFO) &&
+        log.reserve(64)) {
+      log  = F("dotmatrix: bar Width: ");
+      log += barWidth;
+      log += F(" low: ");
+      log += zstruct._lower;
+      log += F(" high: ");
+      log += zstruct._upper;
+    }
+    #  endif // ifdef P104_DEBUG
+    modulesOnOff(zstruct._startModule, zstruct._startModule + zstruct.size - 1, MD_MAX72XX::MD_OFF); // Stop updates on modules
+    P->setIntensity(zstruct.zone - 1, zstruct.brightness); // don't forget to set the brightness
+    uint8_t row = 0;
+
+    if ((barGraphs.size() == 3) || (barGraphs.size() == 5) || (barGraphs.size() == 6)) { // Center within the rows a bit
+      for (; row < (barGraphs.size() == 5 ? 2 : 1); row++) {
+        for (uint8_t col = zstruct._lower; col <= zstruct._upper; col++) {
+          pM->setPoint(row, col, false);                                                 // all off
+
+          if (col % 16 == 0) { delay(0); }
+        }
+        delay(0); // Leave some breathingroom
+      }
+    }
+
+    for (auto it = barGraphs.begin(); it != barGraphs.end(); ++it) {
+      if (essentiallyEqual(it->min, 0.0)) {
+        pixTop    = zstruct._lower - 1 + (((zstruct._upper + 1) - zstruct._lower) / it->max) * it->value;
+        pixBottom = zstruct._lower - 1;
+        zeroPoint = 0;
+      } else {
+        if (definitelyLessThan(it->min, 0.0) && definitelyGreaterThan(it->max, 0.0)) { // Zero-point is used
+          zeroPoint = (it->min * -1.0) / ((it->max - it->min) / (1.0 * ((zstruct._upper + 1) - zstruct._lower)));
```

Comment:
Then initialization of the struct is done during reading the arguments. The attribute is either reset to 0 or set from the argument. But I've explicitly initialized them now.
When setting up a barGraph with those kind of arguments, they might 'deserve' it to 'crash' 😛, but to prevent that I've added another check for 'max - min > 0.01' (they are already in the right 'order', max being greater than min, checked at about 70 lines up).